### ISSUE TITLE
Exclude private subnets when provisioning in AWS

### ISSUE THREAD

**Issue by peterschmidt85** [dstackai/dstack#908]
Currently, `dstack` uses the first subnet it finds. If the VPC contains private subnets, this causes provisioning logic to fail and the run to hang while attempting to connect via SSH.:

```
DEBUG 2024-02-14T13:28:09.616 dstack._internal.core.services.ssh.tunnel SSH tunnel failed: b'ssh: connect to host 34.229.201.59 port 22: Operation timed out\r\n'
```

**Comment by peterschmidt85** [comment]
@r4victor Private subnets should be excluded even if the VPC is not specified in `~/.dstack/server/config.yml`. 

Currently, if the default VPC has private subnets and the VPC name is not specified, runs will fail.

**Comment by peterschmidt85** [comment]
Should be fixed in `0.15.2rc2`
